src/interpreter/executor/interpret.js

```diff
@@ -9,6 +9,9 @@ let output = [];
 
 // Setup for when we initially check syntax
 export function evaluate(inputCode) {
+  // Reset the previous output list.
+  output = [];
```

Comment:
This isn't necessary I don't think.  We have a resetInterpreter in `interpreter/executer/index.js` which sets output to null, can we just call that function if the interpreter errors similarly how we do if it is successful

Reply:
That's what's causing the error. resetInterpreter isn't doing what it's supposed to.

Reply:
The problem right now is that `evaluate()` in `interpreter/executor/interpret.js` is not a pure function and it uses the `output` variable in `interpreter/executor/interpret.js` to build the steps for the eventual return value in `evaluate()`. On a syntax error, the output in `interpret.js` never gets reset, causing the hangup.

In the latest update I've renamed the variable in `index.js` to `expressions` to prevent the confusion.

There are 3 solutions
1. This diff: Reset the `interpret.js`'s output variable before execution
2. Make `index.js` and `interpret.js` use the same variable in `state.js` or something.
3. In `interpret.js`, remove the global variable `output` and change the functions to use a local `output` in all functions so it isn't using a global variable to build the program.

As of now, `index.js` runs evaluate and keeps a copy of the output, and handles the individual steps. The current solution does not fix the duplication of the output from `interpret.js`, but I think the current interface and sandboxing used in `index.js` available for the frontend is ideal (meaning solution 2 is not as nice).

I am in favor of solution 1 (this diff) as of now. It's the least disruptive change to the state of the project to fix the problem. I'd like solution 3 too, but I think there's a lot of code to change to support that.

Reply:
Solution 3 isn't really viable I don't think. The only reason our output is global is so that I can push expressions on during our recursion for evaluating chaining assignments etc. Is there a reason that this issue happens for error but not for regular code? If output is never cleared why don't we have an issue for just subsequent runs?